pkg/operation/garden/garden.go

```diff
@@ -187,8 +179,32 @@ func ReadGardenSecrets(k8sInformers kubeinformers.SharedInformerFactory) (map[st
 			monitoringSecret := secret
 			secretsMap[common.GardenRoleGlobalMonitoring] = monitoringSecret
 			logger.Logger.Infof("Found monitoring basic auth secret %s.", secret.Name)
-		} else {
-			logger.Logger.Info("No monitoring basic auth secret found.")
+		}
+	}
+
+	selectorGardenRole, err := labels.Parse(v1alpha1constants.GardenRole)
+	if err != nil {
+		return nil, err
+	}
+
+	secretsGardenRole, err := k8sInformers.Core().V1().Secrets().Lister().Secrets(v1alpha1constants.GardenNamespace).List(selectorGardenRole)
```

Comment:
Hm, instead of listing twice with two different selectors, can we simply combine the selector and only have one list call?

Reply:
When doing this, you could also allow for the other secrets to use the new `gardener.cloud/role `label (the old `garden.sapcloud.io/role` is deprecated, but we have not yet switched, so we should enable the new `gardener.cloud/role` label in order to remove the deprecated one, eventually).

Reply:
I don't think it's possible to build the selector in a way to match `gardener.cloud/role` OR `garden.sapcloud.io/role`, so we will have to call list twice.